src/main/java/org/cbioportal/web/SessionServiceController.java

```diff
@@ -232,6 +244,29 @@ public ResponseEntity<Session> getSession(@PathVariable Session.SessionType type
         }
     }
 
+    /**
+     * This method populates the `virtualStudyData` object with a new set of sample IDs retrieved as the result of executing a query based on virtual study view filters.
+     * It first applies the filters defined within the study view, runs the query to fetch the relevant sample IDs, and then updates the virtualStudyData to reflect these fresh results.
+     * This ensures that the virtual study contains the latest sample IDs.
+     * @param virtualStudyData
+     */
+    private void populateVirtualStudySamples(VirtualStudyData virtualStudyData) {
+        List<SampleIdentifier> sampleIdentifiers = studyViewFilterApplier.apply(virtualStudyData.getStudyViewFilter());
+        Map<String, Set<String>> sampleIdsByStudyId = sampleIdentifiers
+            .stream()
+            .collect(
+                Collectors.groupingBy(
+                    SampleIdentifier::getStudyId,
+                    Collectors.mapping(SampleIdentifier::getSampleId, Collectors.toSet())));
+        Set<VirtualStudySamples> virtualStudySamples = sampleIdsByStudyId.entrySet().stream().map(entry -> {
+            VirtualStudySamples vss = new VirtualStudySamples();
+            vss.setId(entry.getKey());
+            vss.setSamples(entry.getValue());
+            return vss;
+        }).collect(Collectors.toSet());
```

Comment:
Opinion, but I would extract this into a function called `extractVirtualStudySamples` or something similar.

Reply:
@haynescd Please have a look https://github.com/cBioPortal/cbioportal/pull/11040/commits/eda7367cf717a30aa1c6a098f407c663015f5757